Docs: Add xy chart docs (#81980)
grafana/grafana: docs/sources/panels-visualizations/visualizations/xy-chart/index.md

---
canonical: https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/xy-chart/
keywords:
  - grafana
  - chart
  - xy chart
  - documentation
  - guide
  - graph
labels:
  products:
    - cloud
    - enterprise
    - oss
description: Configure options for Grafana's xy chart
title: XY chart
weight: 100
---

# XY chart

XY charts provide a way to visualize...

<!--add brief description here-->

![An xy chart showing height weight distribution](screenshot-xy-chart-v10.4.png)

<!--add more ideal use cases/examples here-->

## Supported data formats

> TBA

Author: ```suggestion
For an xy chart, you need to provide any type of tabular data. This type of visualization doesn't require time data.
```

## XY chart options

### Series mapping

Set how series data is mapped in the visualization.

> <!--i don't understand these descriptions-->

Author: ```suggestion
<!--to be refined-->
```

> - **Table** - Plot values within a single table result.

Author: ```suggestion
- **Auto** - Automatically generates series based on field selection across multiple data frames.
```

> - **Manual** - Construct values from any result. You can add multiple series for which you'll set the **X-field** and **Y-field** options. When you choose this option, the **Point color** and **Point size** fields are also displayed.

Author: ```suggestion
- **Manual** - Construct values from any result. You can explicitly define multiple series for which you'll set the **X-field** and **Y-field** options, as well as the **Point color** and **Point size** rules
```

### Data

<!--what is the scenario where you need to choose a data set?-->

> Select your data set. This option is only available when you set **Series mapping** to **Table** mode.

Author: ```suggestion
Select your data set. This option is only available when you set **Series mapping** to **Manual** mode.
```

### X-field

Select which field the x-axis represents.

<!--can we make clear here how this is different when you map your series by table or manual; there seem to be more fields to choose from on manual mode-->

### Y-fields

Select which fields the y-axis represents.

> **Table** mode - When you set the series mapping to **Table** mode, you can only update the x-field and the y-fields are automatically assigned based on that selection. In this mode you can enable and disable the **Y-field**.

Author: ```suggestion
**Table** mode - When you set the series mapping to **Table** mode, you can only update the x-field and the y-fields are automatically assigned based on that selection. In this mode you can enable and disable the **Y-field**.

For example, if your data source is a table with five columns, you may want to disable one or more of the columns so that they're not included in the visualization.
```

> <!--why would you want to disable the y-field-->

Author: ```suggestion
```

---
canonical: https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/xy-chart/
keywords:
  - grafana
  - chart
  - xy chart
  - documentation
  - guide
  - graph
labels:
  products:
    - cloud
    - enterprise
    - oss
description: Configure options for Grafana's xy chart
title: XY chart
weight: 100
---

# XY chart

> XY charts provide a way to visualize arbitrary x and y values in a graph so that you can easily show the relationship between two variables.

Author: ```suggestion
XY charts provide a way to visualize arbitrary x and y values in a graph so that you can easily show the relationship between two variables. XY charts are typically used to create scatter plots. You can also use them to create bubble charts where size fields are mapped to the points or bubbles in the visualization. 
```

![An xy chart showing height weight distribution](screenshot-xy-chart-v10.4.png)

## Supported data formats

> You can use any type of tabular data in an xy chart. This type of visualization doesn't require time data.

Reviewer: I'd slightly rephrase this. Two numeric fields are required for an XY chart so it can't support any tabular data (most of it though 😄).

> You can use any type of tabular data in an xy chart. This type of visualization doesn't require time data.

Author: ```suggestion
You can use any type of tabular data with at least two numeric fields in an xy chart. This type of visualization doesn't require time data.
```

## XY chart options

### Series mapping

Set how series data is mapped in the chart.

- [Auto](#auto-series-mapping-options) - Automatically generates series from all available data frames (or datasets). You can filter to select only one frame.
- [Manual](#manual-series-mapping-options) - Explicitly define the series by selecting from available data frames.

Depending on your series mapping selection, the **Frame**, **X-field**, and **Y-field(s)** options differ. The [Auto](#auto-series-mapping-options) and [Manual](#auto-series-mapping-options) series mapping sections describe those different settings.

#### Auto series mapping options

When you select **Auto** as your series mapping mode, the following options become available:

- [Frame](#frame)
- [X-field](#x-field)
- [Y-fields](#y-fields)

##### Frame

> By default, xy chart displays all data frames. You can filter to select only one frame.

Reviewer: Not sure if XY is always capitalized, seems like it based on the page title?

Author: So for the page title, I've capitalized it because it would be awkward not to. But if it's not at the beginning of a sentence or in a title, then lower case it is.

##### X-field

Select which field or fields x represents. By default, this is the first number field in each data frame. For example, you enter the following CSV content:

| a   | b   | c   |
| --- | --- | --- |
| 0   | 0   | 0   |
| 1   | 1   | 9   |
| 2   | 2   | 4   |

In the resulting chart, x-fields are generated from the values in column "a" unless you define it differently.

##### Y-fields

The series of the chart are generated from the y-fields. After the x-field is set, by default, all the remaining number fields in the data frame are designated as the y-fields.

You can select one y-field to include or you can use [overrides][Configure field overrides] to exclude y-fields individually. To disable y-fields individually, add an override with the following properties for each y-field you want removed:

- Override type: **Fields with name**
- Override property: **Series > Hide in area**
- Area: **Viz**

#### Manual series mapping options

When you select **Manual** as your series mapping mode, the following options become available:

- **Frame** - Select your data frame or dataset. You can add as many data frames as you want.

> - **X-field** - Select which field x represents.

Reviewer: I'm not sure what we normally do in this situation but it could be helpful to emphasize `x` and `y` in the document. There were a few times I almost read over it but that may just be me

Author: Let's leave it for now, since at the moment, it's following our style rules. If we get feedback that indicates we need to make a change, we'll go from there.
